tests/integration/test_integration.py

```diff
@@ -212,6 +215,66 @@ async def test_given_tls_requirer_is_deployed_with_3_units_and_related_when_prov
             "utf-8"
         ).strip("\n")
 
+        await ops_test.model.applications[TLS_REQUIRER_CHARM_NAME].destroy_relation(
+            local_relation="certificates",
+            remote_relation=f"{APPLICATION_NAME}:certificates",
+            block_until_done=True,
+        )
+        await ops_test.model.applications[APPLICATION_NAME].scale(1)
+
+    async def test_given_certificate_transfer_requirer_related_when_config_changed_then_certificate_is_passed(  # noqa: E501
+        self, ops_test: OpsTest
+    ):
+        assert ops_test.model
+
+        any_app_name = "any-cert-transfer-requirer"
+        cert_transfer_lib_url = "https://github.com/canonical/certificate-transfer-interface/raw/main/lib/charms/certificate_transfer_interface/v1/certificate_transfer.py"  # noqa: E501
+        cert_transfer_lib = requests.get(cert_transfer_lib_url, timeout=10).text
```

Comment:
This makes me super uncomfortable, and I'm not sure of all the reasons yet.

At least one concern is that this is non-deterministic.

Can this just grab the lib that is vendored with the charm? Why do we need to download it?

Reply:
You are right, I used the examples from AnyCharm's documentation and other charms that use it, but reading the vendored lib is a better idea. I'll add that.

Reply:
Using the vendored lib instead.